src/Core/Services/Implementations/NotificationHubPushNotificationService.cs

```diff
@@ -255,16 +264,32 @@ private string BuildTag(string tag, string identifier)
 
     private async Task SendPayloadAsync(string tag, PushType type, object payload)
     {
-        var outcome = await _client.SendTemplateNotificationAsync(
-            new Dictionary<string, string>
-            {
-                { "type",  ((byte)type).ToString() },
-                { "payload", JsonSerializer.Serialize(payload) }
-            }, tag);
-        if (_globalSettings.NotificationHub.EnableSendTracing)
+        var tasks = new List<Task<NotificationOutcome>>();
+        foreach (var client in _clients)
         {
-            _logger.LogInformation("Azure Notification Hub Tracking ID: {id} | {type} push notification with {success} successes and {failure} failures with a payload of {@payload} and result of {@results}",
-                outcome.TrackingId, type, outcome.Success, outcome.Failure, payload, outcome.Results);
+            var task = client.SendTemplateNotificationAsync(
+                new Dictionary<string, string>
+                {
+                    { "type",  ((byte)type).ToString() },
+                    { "payload", JsonSerializer.Serialize(payload) }
+                }, tag);
+            tasks.Add(task);
+        }
+
+        await Task.WhenAll(tasks);
+
+        if (!_enableTracing)
```

Comment:
🎨 I'd rather flip this and do the tracing when it's true, not short-circuit like this. Will be easier to make mistakes in the future with downstream logic when it's like this.

Reply:
Funny because I had it like you suggest first, but I usually prefer to short circuit methods instead of nested deep if conditions, so I changed it to this. But in this case you are probably right for readability reasons.